packages/flutter_tools/lib/src/windows/msbuild_utils.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file
+
+import 'package:xml/xml.dart' as xml;
+
+import '../base/file_system.dart';
+import '../base/io.dart';
+import '../base/platform.dart';
+import '../base/process_manager.dart';
+
+/// Returns the path to an installed vcvars64.bat script if found, or null.
+Future<String> findVcvars() async {
+  final String programDir = platform.environment['PROGRAMFILES(X86)'];
+  final String pathPrefix = fs.path.join(programDir, 'Microsoft Visual Studio');
+  const String vcvarsScriptName = 'vcvars64.bat';
+  final String pathSuffix =
+      fs.path.join('VC', 'Auxiliary', 'Build', vcvarsScriptName);
+  final List<String> years = <String>['2017', '2019'];
+  final List<String> flavors = <String>[
+    'Community',
+    'Professional',
+    'Enterprise',
+    'Preview'
+  ];
+  for (final String year in years) {
+    for (final String flavor in flavors) {
+      final String testPath =
+          fs.path.join(pathPrefix, year, flavor, pathSuffix);
+      if (fs.file(testPath).existsSync()) {
+        return testPath;
+      }
+    }
+  }
+
+  // If it can't be found manually, check the path.
+  final ProcessResult whereResult = await processManager.run(<String>[
+    'where.exe',
+    vcvarsScriptName,
+  ]);
+  if (whereResult.exitCode == 0) {
+    return whereResult.stdout.trim();
+  }
+
+  return null;
+}
+
+/// Writes a property sheet (.props) file to expose all of the key/value
+/// pairs in [variables] as enivornment variables.
+Future<void> writePropertySheet(
```

Comment:
We should add a test that the xml has the correct structure. This should be fairly easy to read from the MemoryFileSystem

Reply:
Done. (We may want to add more over time, but there's some initial validation now.)